backends/dpdk/dpdkAsmOpt.h

```diff
@@ -174,18 +174,23 @@ class ShortenTokenLength : public Transform {
     // 1.30.30 => 63(including dot(.))
     // if id name less than allowedLength keep it same
     cstring shortenString(cstring str, size_t allowedLength = 60) {
-        if (str.size() <= allowedLength)
+        // this for getting rid of suffixes in NoAction_0 or NoAction_1
+        if (str.size() > 8 && str.substr(0, 8) == "NoAction") {
```

Comment:
This string manipulation can be avoided if you check originalName in caller. Can we do that?

Reply:
I'm not sure dpdk ir has that

Reply:
(snip)
```
class DpdkAction {
    optional Annotations        annotations = Annotations::empty;
    inline IndexedVector<DpdkAsmStatement> statements;
    inline ID name;
...
}

``` 

   name.originalName should be available.